packages/fuel-gauge/src/contract-factory.test.ts

```diff
@@ -245,21 +253,232 @@ describe('Contract Factory', () => {
+    using launched = await launchTestNode();
+
+    const {
+      wallets: [wallet],
+    } = launched;
+
+    const factory = new ContractFactory(LargeContractFactory.bytecode, LargeContract.abi, wallet);
+    const chunkSizeOverride = 2;
+
+    await expectToThrowFuelError(
+      () => factory.deployContractAsBlobs<LargeContract>({ chunkSizeOverride }),
+      new FuelError(
+        ErrorCode.INVALID_CHUNK_SIZE_OVERRIDE,
+        'Chunk size override must be between 0 and 1'
+      )
+    );
+  });
+
+  it('deploys a small contract via blobs', async () => {
+    using launched = await launchTestNode();
+
+    const {
+      wallets: [wallet],
+    } = launched;
+
+    const factory = new ContractFactory(
+      ConfigurableContractFactory.bytecode,
+      ConfigurableContract.abi,
+      wallet
+    );
+
+    const deploy = await factory.deployContractAsBlobs();
+    const { contract } = await deploy.waitForResult();
+
+    const call = await contract.functions.echo_u8().call();
+    const { value } = await call.waitForResult();
+    expect(value).toBe(10);
+  }, 15000);
+
+  it('deploys a small contract via deploy entrypoint', async () => {
+    using launched = await launchTestNode();
+
+    const {
+      wallets: [wallet],
+    } = launched;
+
+    const deploy = await ConfigurableContractFactory.deploy(wallet);
+    const { contract } = await deploy.waitForResult();
+
+    const call = await contract.functions.echo_u8().call();
+    const { value } = await call.waitForResult();
+    expect(value).toBe(10);
+  });
+
+  it('deploys a large contract via deploy entrypoint', async () => {
+    using launched = await launchTestNode({
+      providerOptions: {
+        cacheUtxo: -1,
+      },
+    });
+
+    const {
+      wallets: [wallet],
+    } = launched;
+
+    const deploy = await LargeContractFactory.deploy(wallet);
+    const { contract } = await deploy.waitForResult();
+
+    const call = await contract.functions.something().call();
+    const { value } = await call.waitForResult();
+    expect(value.toNumber()).toBe(1001);
+  }, 15000);
+
+  it('should not deploy large contract with invalid balance', async () => {
+    using launched = await launchTestNode({
+      providerOptions: {
+        cacheUtxo: -1,
+      },
+      walletsConfig: {
+        amountPerCoin: 0,
+      },
+    });
+
+    const {
+      wallets: [wallet],
+    } = launched;
+
+    const factory = new ContractFactory(LargeContractFactory.bytecode, LargeContract.abi, wallet);
+
+    await expectToThrowFuelError(
+      () => factory.deploy({ maxFee: 0 }),
+      new FuelError(ErrorCode.FUNDS_TOO_LOW, 'Insufficient balance to deploy contract.')
+    );
+  });
+
+  it('deploys a contract via blobs [configurables]', async () => {
+    using launched = await launchTestNode();
+
+    const {
+      wallets: [wallet],
+    } = launched;
+
+    const factory = new ContractFactory(
+      ConfigurableContractFactory.bytecode,
+      ConfigurableContract.abi,
```

Comment:
Do we want to be testing that the `LargeContract` can be deployed with configurables?

Reply:
IMO as we are enforcing blob deployment via `deployContractAsBlobs` it doesn't make much difference which contract is used. The configurable contract bytecode will still be configured and deployed as blob, then accessed by the loader contract which takes the blob ID. The only difference will be in the number of blobs, the loader contract loops through, the logic remains the same.